packages/@o3r/localization/README.md

```diff
@@ -1,3 +1,703 @@
+export function localizationConfigurationFactory(): LocalizationConfiguration {
+  return {
+    ...DEFAULT_LOCALIZATION_CONFIGURATION,
+    supportedLocales: ['en-GB', 'fr-FR', 'ar-AR'],
+    fallbackLocalesMap: {
+      'en-CA': 'en-GB',
+      'en-US': 'en-GB',
+      'de': 'fr-FR',
+      'zh': 'en-GB'
+    },
+    fallbackLanguage: 'en-GB',
+    bundlesOutputPath: environment.LOCALIZATION_BUNDLES_OUTPUT,
+    useDynamicContent: true,
+    // dummy example
+    enableTranslationDeactivation: window.location.search.indexOf('debug=true') >= 0
+  };
+}
+
+@NgModule({
+  imports: [LocalizationModule],
+  providers: [{provide: LOCALIZATION_CONFIGURATION_TOKEN, useFactory: localizationConfigurationFactory}]
+})
+export class AppModule {
+}
+````
+
+Example of usage in a debug component:
+
+````typescript
+// Component class
+@Component({
+  selector: 'debug',
+  templateUrl: './debug.template.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DebugComponent {
+  constructor(private localizationService: LocalizationService) {}
+
+  public toggleTranslation() {
+    this.localizationService.toggleShowKeys();
+  }
+}
+````
+
+````angular2html
+<!-- Component template -->
+<div>
+  <h3>Debug:</h3>
+  <button (click)="toggleTranslation()">Toggle translation</button>
+</div>
+````
+
+### Bootstrap: enable debug mode to display ``<key> - <translation>``
+
+You can enable debug mode by setting debugMode property of LocalizationConfiguration (by default the value of debugMode is false)
+
+```typescript
+
+registerLocaleData(localeEN, 'en-EN');
+registerLocaleData(localeFR, 'fr');
+registerLocaleData(localeAR, 'ar');
+
+@NgModule({
+  imports: [
+  LocalizationModule.forRoot({
+    language: 'fr',
+    endPointUrl: 'http://example.com',
+    bundlesOutputPath: config.LOCALIZATION_BUNDLES_OUTPUT,
+    supportedLocales: ['en-EN', 'fr', 'ar'],
+    debugMode: true
+  }),
+  ...
+})
+
+```
+
+By doing this all your translations will be prefixed by the corresponding localization key. That way you can easily map the text to the key.
+
+Examples:
+
+- [simpleHeader.airline] My Airline
+
+- [simpleHeader.motto] Let's shape the future of travel
+
+## How to properly create keys?
+
+It's always good to stick to some naming conventions.
+
+For components we prefix each key by component selector to ensure cross components keys uniqueness.
+
+For example if your `SimpleHeaderPresComponent` has `o3r-simple-header-pres` selector then your keys may look like
+
+```typescript
+
+{
+  "o3r-simple-header-pres.motto": {
+    "description": "airline motto phrase",
+    "defaultValue": "Let's shape the future of travel"
+  },
+  "o3r-simple-header-pres.airline": {
+    "description": "airline name",
+    "defaultValue": "My Airline"
+  }
+}
+
+```
+
+## Intelligent fallback support
+
+The fallback hierarchy is been added to the localization service which explained in detail below. As the last option, the default fallback language picked as usual.
```

Comment:
**has** been added
which **is** explained
As a last resort, the default fallback language is picked as usual.